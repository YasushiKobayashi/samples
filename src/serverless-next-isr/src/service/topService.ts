import * as prismic from '@prismicio/client'

import { fetchCategories, fetchPosts } from '@/repository/prismic/client'

export const topService = async (client: prismic.Client) => {
  const posts = fetchPosts(client)
  const categories = fetchCategories(client)
  const res = await Promise.all([posts, categories])
  return { posts: res[0], categories: res[1] }
}
