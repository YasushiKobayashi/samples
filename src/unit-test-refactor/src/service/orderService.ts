import Router from 'next/router'

import { CartType } from '@/models/cart'
import { UserType } from '@/models/user'
import { orderPost } from '@/request/cart'

export const submitOrder = async (cartId: CartType['id'], user: UserType) => {
  await orderPost(cartId)
  const path = user.orders.length > 0 ? '/thanks_repeater' : '/thanks_new_user'
  Router.push(path)
}
