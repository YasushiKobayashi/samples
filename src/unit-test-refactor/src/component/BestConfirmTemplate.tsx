import * as React from 'react'

import { CartType, getPaymentAmount, getShippingFee, getTotalPrice } from '@/models/cart'
import { UserType } from '@/models/user'

import { CartPriceInfo } from '@/component/CartPriceInfo'
import { submitOrder } from '@/service/orderService'
import { DeliverInfo } from './DeliverInfo'
import { CartDetailInfo } from './CartDetailInfo'

interface Props {
  cart: CartType
  user: UserType
}

export const BestConfirmTemplate: React.FC<Props> = ({ cart, user }) => {
  const submit = async (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault()
    try {
      submitOrder(cart.id, user)
    } catch (err) {
      console.error(err)
    }
  }

  const totalPrice = getTotalPrice(cart)
  const shippingFee = getShippingFee(totalPrice)
  const paymentAmount = getPaymentAmount(totalPrice, shippingFee)

  return (
    <div>
      <form onSubmit={submit}>
        <DeliverInfo user={user} />
        <CartDetailInfo cart={cart} />
        <CartPriceInfo shippingFee={shippingFee} paymentAmount={paymentAmount} />
        <button type="submit">購入する</button>
      </form>
    </div>
  )
}
