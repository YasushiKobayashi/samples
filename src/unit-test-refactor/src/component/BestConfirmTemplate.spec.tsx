import * as React from 'react'
import { cleanup, fireEvent, render } from '@testing-library/react'

import { submitOrder } from '@/service/orderService'

import { BestConfirmTemplate } from './BestConfirmTemplate'

describe('BestConfirmTemplate', () => {
  beforeEach(() => {
    /* eslint-disable no-import-assign,@typescript-eslint/ban-ts-comment */
    // @ts-ignore
    submitOrder = jest.fn()
  })

  afterEach(() => {
    cleanup()
  })

  it('購入まで', () => {
    const props = {
      user: {
        id: 1,
        name: 'Yasushi Kobayashi',
        address: {
          id: 1,
          postCode: '162-0846',
          prefecture: '東京都',
          city: '新宿区',
          address: '市谷左内町21-13',
          mansion: null,
        },
        orders: [
          {
            id: 1,
            userId: 1,
          },
        ],
      },
      cart: {
        id: 1,
        userId: 1,
        details: [
          {
            productId: 1,
            quantity: 1,
            product: {
              id: 1,
              name: 'TypeScript入門',
              price: 3000,
            },
          },
          {
            productId: 2,
            quantity: 2,
            product: {
              id: 2,
              name: 'Python入門',
              price: 4000,
            },
          },
        ],
      },
    }
    const { container, asFragment, getByText } = render(<BestConfirmTemplate {...props} />)

    expect(container.textContent).toContain('配送先情報')
    expect(container.textContent).toContain('カート情報')
    expect(container.textContent).toContain('金額情報')

    expect(submitOrder).toHaveBeenCalledTimes(0)

    fireEvent.click(getByText('購入する'))
    expect(submitOrder).toHaveBeenCalledTimes(1)
    expect(submitOrder).toHaveBeenCalledWith(props.cart.id, props.user)

    expect(asFragment()).toMatchSnapshot()
  })
})
