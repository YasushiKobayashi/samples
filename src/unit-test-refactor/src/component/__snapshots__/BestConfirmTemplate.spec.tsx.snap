// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BestConfirmTemplate 購入まで 1`] = `
<DocumentFragment>
  <div>
    <form>
      <h2>
        配送先情報
      </h2>
      <p>
        Yasushi Kobayashi様
      </p>
      <p>
        〒162-0846 東京都新宿区市谷左内町21-13
      </p>
      <h2>
        カート情報
      </h2>
      <ul>
        <li>
          <p>
            商品名:TypeScript入門
          </p>
          <p>
            数量:1個
          </p>
          <p>
            価格:3,000円
          </p>
        </li>
        <li>
          <p>
            商品名:Python入門
          </p>
          <p>
            数量:2個
          </p>
          <p>
            価格:4,000円
          </p>
        </li>
      </ul>
      <div>
        <h2>
          金額情報
        </h2>
        <h3>
          送料
        </h3>
        <p>
          無料
        </p>
        <h3>
          合計金額
        </h3>
        <p>
          11,000円
        </p>
      </div>
      <button
        type="submit"
      >
        購入する
      </button>
    </form>
  </div>
</DocumentFragment>
`;
